src/js/server/schemaMiddleware.js

```diff
@@ -0,0 +1,172 @@
+/*
+
+    "Gatekeeper" middleware that rejects any request whose JSON payloads are not valid. See this component's
+    documentation for more details:
+
+    https://github.com/the-t-in-rtf/gpii-json-schema/blob/GPII-1336/docs/middleware.md
+
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("./schemaHandler");
+require("../common/hasRequiredOptions");
+
+fluid.defaults("gpii.schema.middleware.handler", {
+    gradeNames: ["gpii.schema.handler"],
+    invokers: {
+        handleRequest: {
+            func: "{that}.sendResponse",
+            args: [400, "{that}.options.body"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.schema.middleware");
+
+gpii.schema.middleware.rejectOrForward  = function (that, req, res, next) {
+    var toValidate = fluid.model.transformWithRules(req, that.options.rules.requestContentToValidate);
+    var results = that.validator.validate(that.options.schemaKey, toValidate);
+    if (results) {
+        var transformedResults = fluid.model.transformWithRules(results, that.options.rules.validationErrorsToResponse);
+        that.events.onInvalidRequest.fire(req, res, transformedResults);
+    }
+    else {
+        next();
+    }
+};
+
+/*
+
+    The `gpii.express.middleware` that fields invalid responses itself and passes valid ones through to the `next`
+    Express router or middleware function.
+
+ */
+fluid.defaults("gpii.schema.middleware", {
+    gradeNames: ["gpii.express.middleware", "gpii.express.requestAware.base", "gpii.hasRequiredOptions"],
+    requiredFields: {
+        "schemaPath": true,
+        "schemaKey": true
+    },
+    responseSchemaKey: "message.json",
+    responseSchemaUrl: "http://terms.raisingthefloor.org/schema/message.json",
+    messages: {
+        error: "The JSON you have provided is not valid."
+    },
+    rules: {
+        validationErrorsToResponse: {
+            "ok": {
+                literalValue: false
+            },
+            "message": {
+                literalValue: "{that}.options.messages.error"
+            },
+            "fieldErrors": ""
+        }
+    },
+    events: {
+        "onInvalidRequest": null
+    },
+    handlerGrades: ["gpii.schema.middleware.handler"],
+    dynamicComponents: {
+        requestHandler: {
```

Comment:
This is another example of router functionality that belongs upstairs in the router. We shouldn't need to construct an entire handler component because we wish to fire an error.